code/game/machinery/computer/security.dm

```diff
@@ -319,237 +201,254 @@
-					if(istype(active1, /datum/data/record) && L.Find(rank))
-						var/list/buttons = list()
-						for(var/rank in GLOB.joblist)
-							buttons[++buttons.len] = list("name" = rank, "icon" = null, "href" = "rank=[rank]", "status" = (active1.fields["rank"] == rank ? "selected" : null))
-						setTemp("<h3>Rank</h3>", buttons)
-					else
-						setTemp("<h3 class='bad'>You do not have the required rank to do this!</h3>")
-				if("species")
-					if(istype(active1, /datum/data/record))
-						var/t1 = copytext(trim(sanitize(input("Please enter race:", "General records", active1.fields["species"], null) as message)), 1, MAX_MESSAGE_LEN)
-						if(!t1 || ..() || active1 != a1)
-							return 1
-						active1.fields["species"] = t1
-	return 1
-
-/obj/machinery/computer/secure_data/proc/setTemp(text, list/buttons = list())
-	temp = list("text" = text, "buttons" = buttons, "has_buttons" = buttons.len > 0)
-
-/* Proc disabled due to BYOND Issue
-
-/obj/machinery/computer/secure_data/proc/create_record_photo(datum/data/record/R)
-	// basically copy-pasted from the camera code but different enough that it has to be redone
-	var/icon/photoimage = get_record_photo(R)
-	var/icon/small_img = icon(photoimage)
-	var/icon/tiny_img = icon(photoimage)
-	var/icon/ic = icon('icons/obj/items.dmi',"photo")
-	var/icon/pc = icon('icons/obj/bureaucracy.dmi', "photo")
-	small_img.Scale(8, 8)
-	tiny_img.Scale(4, 4)
-	ic.Blend(small_img, ICON_OVERLAY, 10, 13)
-	pc.Blend(tiny_img, ICON_OVERLAY, 12, 19)
-
-	var/datum/picture/P = new()
-	P.fields["name"] = "File Photo - [R.fields["name"]]"
-	P.fields["author"] = "Central Command"
-	P.fields["icon"] = ic
-	P.fields["tiny"] = pc
-	P.fields["img"] = photoimage
-	P.fields["desc"] = "You can see [R.fields["name"]] on the photo."
-	P.fields["pixel_x"] = rand(-10, 10)
-	P.fields["pixel_y"] = rand(-10, 10)
-	P.fields["size"] = 2
-
-	var/obj/item/photo/PH = new/obj/item/photo(loc)
-	PH.construct(P)
-
-*/
-
-/obj/machinery/computer/secure_data/proc/get_record_photo(datum/data/record/R)
-	// similar to the code to make a photo, but of course the actual rendering is completely different
-	var/icon/res = icon('icons/effects/96x96.dmi', "")
-	// will be 2x2 to fit the 2 directions
-	res.Scale(2 * 32, 2 * 32)
-	// transparent background (it's a plastic transparency, you see) with the front and side icons
-	res.Blend(icon(R.fields["photo"], dir = SOUTH), ICON_OVERLAY, 1, 17)
-	res.Blend(icon(R.fields["photo"], dir = WEST), ICON_OVERLAY, 33, 17)
-
-	return res
+					return FALSE
+		else
+			return FALSE
+
+/**
+  * Called when the print record timer finishes
+  */
+/obj/machinery/computer/secure_data/proc/print_record_finish()
+	var/obj/item/paper/P = new(loc)
+	P.info = "<center><b>Security Record</b></center><br>"
+	if(record_general && GLOB.data_core.general.Find(record_general))
+		P.info += {"Name: [record_general.fields["name"]] ID: [record_general.fields["id"]]
```

Comment:
**Please** make sure that you cant set someone's crimes to `<img src="lol">` and then print the paper. TGUI may be sanitized, but paper is not.

Alternatively, make sure all inputs are sanitized regardless.

Reply:
![image](https://user-images.githubusercontent.com/1496804/94126734-45368700-fe58-11ea-82c2-cb849d3f6440.png)